curp/src/server/raw_curp/state.rs

```diff
@@ -87,6 +90,8 @@ impl FollowerStatus {
 pub(super) struct LeaderState {
     /// For each server, the leader maintains its status
     statuses: DashMap<ServerId, FollowerStatus>,
+    /// Leader Transferee
+    leader_transferee: AtomicU64,
```

Comment:
I think it's better to use a lock here to provide synchronization. For example, if the leader observed that it has a transferee during the append entries, right before it will send timeout request, the leader election tick timeout and reset the transferee, the `move_leader` interface will returns an error to the client even when the leader is acutally transfered.

Reply:
make sense